code/socialDistribution/views.py

```diff
@@ -389,14 +390,24 @@ def author(request, author_id):
         author_type = REMOTE
 
     if author_type == LOCAL:
-            posts = author.posts.listed().get_public()  # get public posts only
+        posts = author.posts.listed().get_public()  # get public posts only
 
     else:
         # show public posts only
-        posts = InboxPost.objects.filter(
-            author=author.get_url_id(),
-            visibility=InboxPost.Visibility.PUBLIC
-        )
+
+        request_url = author.get_url_id().strip('/') + "/posts"
+
+        res_code, res_body = api_requests.get(request_url, send_basic_auth_header=True)
+        
+        if res_code == 200 and res_body:
+            for post in res_body["items"]:
+                if post:
+                    makePost(post)
+
+            posts = InboxPost.objects.filter(
+                author=author.get_url_id(),
+                visibility=InboxPost.Visibility.PUBLIC
+            )
```

Comment:
Should this be one indentation level less? It will only be called when the get request succeeds

Reply:
That's a nice catch! Fixed